job_definitions/smoke_tests.yml

```diff
@@ -1,42 +1,86 @@
+    "smoke": variant_bases.smoke|combine(environment_bases["preview"], {
+      "vars": smoke_test_variables["preview"],
+    }),
+    "smoulder": variant_bases.smoulder|combine(environment_bases["preview"], {
+      "vars": smoulder_test_variables["preview"],
+    }),
+  },
+  "staging": {
+    "smoke": variant_bases.smoke|combine(environment_bases["staging"], {
+      "vars": smoke_test_variables["staging"],
+    }),
+    "smoulder": variant_bases.smoulder|combine(environment_bases["staging"], {
+      "vars": smoulder_test_variables["staging"],
+    }),
+  },
+  "production": {
+    "smoke": variant_bases.smoke|combine(environment_bases["production"], {
+      "vars": smoke_test_variables["production"],
+    }),
+    "smoulder": variant_bases.smoulder|combine(environment_bases["production"], {
+      "vars": smoulder_test_variables["production"],
+    }),
+  },
+} %}
 ---
-{% for environment in environments %}
+{% for environment, variants in environments.items() %}
+{% for variant, instance in variants.items() %}
 - job:
-    name: "apps-are-up-{{ environment }}"
-    display-name: "Apps are up - {{ environment }}"
+    name: "{{ instance.slugname }}-{{ environment }}"
+    display-name: "{{ instance.displayname }} - {{ environment }}"
     project-type: pipeline
     description: |
-      Check that the eight Digital Marketplace apps are up and running in the {{ environment|upper }} environment:
-      <ul>
-        <li>API</li>
-        <li>Search API</li>
-        <li>Admin frontend</li>
-        <li>Briefs frontend</li>
-        <li>Brief-responses frontend</li>
-        <li>Buyer frontend</li>
-        <li>Supplier frontend</li>
-        <li>User frontend</li>
-      </ul>
+      Check that all Digital Marketplace apps are {{ instance.assertion_description }} in the {{ environment|upper }} environment
     properties:
       build-discarder:
         days-to-keep: 4
         artifact-days-to-keep: 4
     triggers:
-      - timed: "H/5 * * * *"
+      - timed: "{{ instance.timing }}"
     wrappers:
       - ansicolor
     dsl: |
 
       def notify_slack(icon, status) {
         build job: "notify-slack",
               parameters: [
-                string(name: 'USERNAME', value: 'smoke-tests'),
+                string(name: 'USERNAME', value: '{{ instance.slack_username }}'),
                 string(name: 'ICON', value: icon),
-                string(name: 'JOB', value: "Apps are up - {{ environment }}"),
+                string(name: 'JOB', value: "{{ instance.displayname }} - {{ environment }}"),
                 string(name: 'CHANNEL', value: "#dm-release"),
                 text(name: 'STAGE', value: "{{ environment }}"),
                 text(name: 'STATUS', value: status),
-                text(name: 'URL', value: "<${BUILD_URL}smoke_test_report|${BUILD_DISPLAY_NAME}>")
+                text(name: 'URL', value: "<${BUILD_URL}{{ instance.report_name|lower|replace(" ", "_") }}smoke_test_report|${BUILD_DISPLAY_NAME}>")
```

Comment:
Looking at the xml output for smoulder preview, this line resolves to 

```
text(name: 'URL', value: &quot;&lt;${BUILD_URL}smoulder_test_reportsmoke_test_report|${BUILD_DISPLAY_NAME}&gt;&quot;)
```

The quotes are fine but should `smoulder_test_reportsmoke_test_report` have an underscore in the middle?